### ISSUE TITLE
Document what the --ie is and isn't for, or remove it

### ISSUE THREAD

**Issue by src-code** [yahoo/atomizer#220]
Right now the `--ie` flag fixes a few small things, but doesn't do a whole lot.  It's unclear in the docs what one should expect `--ie` to do, and what it should leave to the user to solve via post-processors.  (I seem to recall there were specific reasons we chose to introduce `--ie` rather than leave it completely up to post-processors, but the reasons now escape me.  Maybe it was because @thierryk felt the fixes were so few and simple, it didn't make sense to require users to integrate a post-processor for them?)

It'd be good to document this, or if we find it's unnecessary to continue supporting `--ie`, we should remove it in favor of documentation that points to a post-processor like PostCSS.

@renatoi @dnewcome @thierryk

**Comment by src-code** [comment]
Inspired by pull #219

**Comment by renatoi** [comment]
I've been feeling lately that sometimes there's too much configuration to be done to properly use atomizer. While we give granularity and freedom to integrate with other tools, it might make sense to ship a complete solution in one package. I believe we should strive for [Convention Over Configuration](https://en.wikipedia.org/wiki/Convention_over_configuration).

**Comment by src-code** [comment]
@renatoi So yours is a vote for automatically using postcss on the output when `--ie` is passed?  Or are you suggesting something else?

**Comment by renatoi** [comment]
Maybe creating a new package that does all the job behind the scenes?

@3den actually talked to me previously about this. He said there was too much stuff to do to properly use atomizer, such as setting up grunt, merge-configs, atomizer itself, postcss, etc.

Having a package that does magic is a package that comes with everything and simply works without too much config, while still keeping the flexibility to be customized.

What are your thoughts?

**Comment by src-code** [comment]
@renatoi I'm all for creating an opinionated package that helps set things up behind the scenes.  Reminds me a little of this: https://github.com/henrikjoreteg/hjs-webpack

**Comment by src-code** [comment]
Though that is a bit outside the scope of this issue!

**Comment by renatoi** [comment]
Yes, sorry to deviate the topic :)

Back to the original IE issue, I believe it is because some post processors don't even care about IE7 compatibility. They go only so far to add possible fallbacks (mostly, browser vendor prefixes, and some variance between different specs/support for a new css 3 feature, this is the case for autoprefixer). I believe the expectation from the PostCSS side is to actually have many developers to work on different small packages to address specific issues for a specific project.

Atomizer itself added a couple of well known hacks for small number of declarations to make them compatible with IE7.

I'd say we should try to keep compatibility with old IE with the --ie flag in Atomizer, everything else is expected to use PostCSS (or a new opinionated atomizer package).

**Comment by dnewcome** [comment]
In talking with @src-code about this, I'm leaning toward removing this. It seems that the original intent was not documented, and this doesn't represent full downlevel IE support.

One strategy is to include fixes for things that would prevent basic accessibility of the page (my recommendation) but as @src-code pointed out, it may be a slippery slope and what constitutes basic accessibility might vary depending on context.

If we decide on a concrete partial IE support strategy going forward we can look at this again, but for now I can't see maintaining this feature without some philosophy on what is supported.

**Comment by src-code** [comment]
@renatoi said...

> I'd say we should try to keep compatibility with old IE with the --ie flag in Atomizer, everything else is expected to use PostCSS (or a new opinionated atomizer package).

I think this is the part that is currently confusing... "keep compatibility with old IE" is open to interpretation.  Currently we patch things like `display: inline-block` and `opacity` when the flag is enabled, but we decided not to fix things like `rgba()`, leaving it up to the developer to address with PostCSS even though a site might be horribly broken without providing fallbacks for those functions.  It seems then like the `--ie` flag is incomplete and unlikely to really solve most people's issues, and only adds to confusion about what it should be fixing.

I'd be inclined to say we should, at most, patch `display: inline-block` by default (no more `--ie` flag) since it's core to layouts, etc, and leave the rest to post-processors.  But even there I could probably be talked into moving it all into post-processors...

**Comment by 3den** [comment]
I think `--ie` should be removed because that can be done by postcss

**Comment by fidian** [comment]
I would agree that these old IE hacks are not being utilized by even old browsers at this point. I've made a pull request that does this: #401.

In case anyone is coming to this issue and is looking to determine what `--ie` enables, here's the list:

* `display: inline-block` automatically adds `zoom: 1` and `*display: inline`
* `overflow:` with any value OTHER THAN `visible` automatically adds `zoom: 1`
* `opacity:` with any value adds `filter: alpha(opacity=...)` with the value converted appropriately
